### ISSUE TITLE
player made buildings keep coming back reappearing after destroyed in MMO

### ISSUE THREAD

**Issue by SourJay** [suriyun-production/mmorpg-kit-docs#780]

**Comment by insthync** [comment]
As I try it was destroyed correctly. About your server,  you uses SQLite or MySQL as database system.

**Comment by SourJay** [comment]
> As I try it was destroyed correctly. About your server, you uses SQLite or MySQL as database system.

SQLite theres some other people in discord that has the same problem

**Comment by khongtuong** [comment]
me too, i'm using mysql first, then try SQLite but still got same issue, in editor it's show error that "you're trying to access object has been destroyed" and it's still there after restart server, plus campfire and chest got this issue when build:

NullReferenceException: Object reference not set to an instance of an object
MultiplayerARPG.StorageEntity.EntityUpdate () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/BuildingSystems/StorageEntity.cs:66)
MultiplayerARPG.CampFireEntity.EntityUpdate () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/BuildingSystems/CampFireEntity.cs:143)
MultiplayerARPG.BaseGameEntity.Update () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/BaseGameEntity.cs:423)

**Comment by insthync** [comment]
You mean it was destroyed by character attacking not from destroying menu, right ?

**Comment by insthync** [comment]
I still can't reproduce it yet.

**Comment by insthync** [comment]
@khongtuong It's not from MMO Mode, so MySQL or SQLite is not relates, I will try an singleplayer/lan mode

**Comment by insthync** [comment]
Okay, for quick fix, you can add codes to `BuildingEntity.cs` -> `OnNetworkDestroy`

```
                CurrentGameManager.DestroyBuildingEntity(Id);
```

![Screenshot_201](https://user-images.githubusercontent.com/2697550/107275312-2988f180-6a84-11eb-9514-a849b96c83c6.png)

**Comment by SourJay** [comment]
im on mmo and it happens when a monster or player destroys a building if i restart the server the building is back there it never goes away perm maybe its the database or server not saving the building being destroyed? whats the sync time? 
look what callepo said in discord:
Callepo [AlphaAF]Today at 12:25 PM
XD
fun fact  its not a save time issue, its been like that in the kit since beginning of times
no one mentioned it
ive seen it myself over year ago
simply didnt care cuase i dont use it and no one seemed to say anything, so i assumed its intentional
just tell suri he change or add option for both if it was intentional

**Comment by insthync** [comment]
Fixed in  1.62f
